programs/voter-stake-registry/src/instructions/extend_stake.rs

```diff
@@ -68,6 +70,28 @@ pub fn extend_stake(
         source_mint_idx,
         VsrError::InvalidMint
     );
+    // check whether target delegate mining is the same as delegate mining from passed context
+    require_eq!(
+        target.delegate,
+        *ctx.accounts.delegate.key,
+        VsrError::InvalidDelegate
+    );
+
+    let (reward_pool, _) = find_reward_pool_address(
```

Comment:
When we extend stake we should already have all the relevant accounts. wdyt of getting the bumps from those to avoid using the `find_` methods?

Reply:
To extract `bump` we need to deserialize `reward_pool`, for example. I doubt it will be cheaper. Also, `delegate_mining` must be found anyway to ensure mining equals the one from a delegate. wdyt?

Reply:
Some unpredictable value typically ranging between 1800 and 7000 CUs per each find_ call. 
An estimate of ~600 CU for loading and deserializing the account of size 4k. 400 basic cost of load + 3*6CU for each extra k after the first one + ~165 CU for the deserialization - this estimate is from chatgpt, so may be totally wrong
I'd say it's an optimization and may be ignored for now. Having it in a backlog for the future us will suffice

Reply:
Hmm. Thanks for such a deep analysis, highly appreciated! I suppose it might be done without breaking backward compatibility so that we may postpone it.

Reply:
https://linear.app/mplx/issue/MTG-438/replace-find-methods-in-staking-contract-with-reading-the-relevant